system-helper/flatpak-system-helper.c

```diff
@@ -2291,15 +2225,15 @@ main (int    argc,
                                   NULL,
                                   NULL);
 
-  ongoing_pulls = g_ptr_array_new ();
+  cache_dirs_in_use = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
 
   /* Ensure we don't idle exit */
   schedule_idle_callback ();
 
   main_loop = g_main_loop_new (NULL, FALSE);
   g_main_loop_run (main_loop);
 
-  g_clear_pointer (&ongoing_pulls, g_ptr_array_unref);
+  g_clear_pointer (&cache_dirs_in_use, g_hash_table_destroy);
```

Comment:
Lock here? I suppose if we get this far and there's another thread still doing its thing, we're already in trouble. But I think it's still the right thing to do.

Reply:
Review accepted and addressed in the latest round of iteration.